hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -536,11 +542,15 @@ public static S3AFileStatus createFileStatus(Path keyPath,
       String owner,
       String eTag,
       String versionId,
-      boolean isCSEEnabled) {
+      S3Client s3Client,
```

Comment:
this isn't really going to trigger an http request is it?
look how we did it in the old code: we just lied about file length in listings

Reply:
The s3 client is added to provide backward compatiblity with v1 client. It is turned off by default and hence no extra http call will be made

Reply:
By default, we stick to old way of simply subtracting 16 bytes from file length